v1/partial/with_configfile.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2018 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package partial
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/google/go-containerregistry/v1"
+	"github.com/google/go-containerregistry/v1/v1util"
+)
+
+// withConfigFile defines the subset of v1.Image used by these helper methods
+// TODO(mattmoor): Consider renaming this to withCore?
+type withConfigFile interface {
```

Comment:
What does this interface provide, since it seems to only be wrapping/renaming the `imageCore` interface?

Reply:
Now `WithConfigFile` is public, but I'm also conflicted (see TODO).